lib/component.cpp

```diff
@@ -72,16 +79,41 @@ void Component::receiveMessage(std::shared_ptr<SpaMessage> message)
   {
     return;
   }
-  else if (message->spaHeader.opcode == op_SPA_SUBSCRIPTION_REQUEST)
+
+  switch (message->spaHeader.opcode)
   {
-    registerSubscriptionRequest(message);
+    case op_SPA_SUBSCRIPTION_REQUEST:
+      registerSubscriptionRequest(message);
+      break;
+
+    case op_SPA_SUBSCRIPTION_REPLY:
+      handleSubscriptionReply(message);
+      break;
+
+    case op_SPA_DATA:
+      handleSpaData(message);
+      break;
+
+    default:
+      std::cout << "Unrecognized message" << std::endl;
+      break;
   }
-  else if (message->spaHeader.opcode == op_SPA_SUBSCRIPTION_REPLY)
+}
+
+void Component::publish()
+{
+  for (auto i = 0u; i < subscribers.size(); ++i)
```

Comment:
Nope! It makes `i` an unsigned integer so it doesn't throw a warning when you compare it with the vector size.